lychee-lib/src/checker/wikilink/mod.rs

```diff
@@ -0,0 +1,78 @@
+//! `WikiLink` Module
+//!
+//! This module contains a Indexer and a Resolver for `WikiLinks`
+//! The Indexer recursively indexes the subdirectories and files in a given base-directory mapping
+//! the filename to the full path
+//! The Resolver looks for found `WikiLinks` in the Index thus resolving the `WikiLink` to a full
+//! filepath
+
+pub(crate) mod index;
+pub(crate) mod resolver;
+
+use crate::ErrorKind;
+use pulldown_cmark::CowStr;
+
+/// In Markdown Links both '#' and '|' act as modifiers
+/// '#' links to a specific Header in a file
+/// '|' is used to modify the link name, a so called "pothole"
+const MARKDOWN_FRAGMENT_MARKER: char = '#';
+const MARKDOWN_POTHOLE_MARKER: char = '|';
+
+/// Clean a `WikiLink` by removing potholes and fragments from a `&str`
+pub(crate) fn wikilink(input: &str, has_pothole: bool) -> Result<CowStr<'_>, ErrorKind> {
+    // Strip potholes (|) from wikilinks
```

Comment:
This confused me to think that the marker is the "pothole" but its the text afterwards.

```suggestion
    // Strip pothole marker (|) and pothole (text after marker) from wikilinks
```